solutions/usaco-531.mdx

```diff
@@ -121,4 +121,103 @@ int main()
 
 </CPPSection>
 
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <cstdio>
+#include <algorithm>
+#include <vector>
+using namespace std;
+
+const int MAX_N = 2000;
+int parent[MAX_N];
+int compsize[MAX_N];
+struct Edge
+{
+	int from, to, weight;
+};
+
+// BeginCodeSnip{Standard DSU operations}
+void init(int n)
+{
+	for (int i = 0; i < n; i++)
+	{
+		parent[i] = i;
+		compsize[i] = 1;
+	}
+}
+
+int find(int a)
+{
+	if (a == parent[a])
+	{
+		return a;
+	}
+	return parent[a] = find(parent[a]);
+}
+
+bool unite(int a, int b)
+{
+	int roota = find(a), rootb = find(b);
+	if (roota == rootb)
+	{
+		return false;
+	}
+	if (compsize[roota] > compsize[rootb])
+	{
+		swap(roota, rootb);
+	}
+	parent[roota] = rootb;
+	compsize[rootb] += compsize[roota];
+	return true;
+}
+// EndCodeSnip
+
+int main()
+{
+	freopen("superbull.in", "r", stdin);
+	freopen("superbull.out", "w", stdout);
+	int n;
+	cin >> n;
+	int ids[MAX_N];
+	for (int i = 0; i < n; i++)
+	{
+		cin >> ids[i];
+	}
+	vector<Edge> edges;
+	for (int i = 0; i < n; i++)
+	{
+		for (int j = i + 1; j < n; j++)
+		{
+			edges.push_back({i, j, ids[i] ^ ids[j]});
+		}
+	}
+	init(n);
+	// sort edges in decreasing order of weight
+	sort(edges.begin(), edges.end(), [](Edge a, Edge b){
+		return a.weight > b.weight;
+	});
+	int eliminated = 0;
+	long long ans = 0;
+	for (Edge e : edges)
```

Comment:
maybe add a short comment "Run Kruskal's on the edges" or something like that idk